internal/datastore/post_db_test.go

```diff
@@ -0,0 +1,74 @@
+package datastore
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/nedrocks/delphisbe/graph/model"
+
+	"github.com/nedrocks/delphisbe/internal/datastore/tests"
+)
+
+type postTestData struct {
+	discussionID string
+	post         model.Post
+}
+
+func TestPostDatastore(t *testing.T) {
+	// Testing variables
+	data := TestData{
+		Discussions:  tests.ValidDiscussions(),
+		Participants: tests.ValidParticipants(),
+		Posts:        tests.ValidPosts(),
+	}
+	postObj := model.Post{
+		ID:            "3031f9ee-fa75-4004-8572-0eba5a0a43b7",
+		DiscussionID:  &tests.Discussion1ID,
+		ParticipantID: &tests.Participant2ID,
+		PostContent: &model.PostContent{
+			ID:      "6b67e749-1edd-4f0f-b8e6-732d28ce3da9",
+			Content: "unit test post",
+		},
+	}
+
+	// Initialize the test DB once per file
+	db, close, err := MakeDatastore(context.Background(), data)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer close()
+
+	// Loop over table tests
+	tests := []struct {
```

Comment:
I know I have one or two tests structured with this format but I find it slightly harder to read than convey. I'm not wedded to one side or the other, but it's harder to convey (harhar) what you're actually testing as much as with Convey.